klyshko-operator/controllers/etcd_keys.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (c) 2022 - for information on the respective copyright owner
+see the NOTICE file and/or the repository https://github.com/carbynestack/klyshko.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package controllers
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/types"
+	"math"
+	"regexp"
+	"strconv"
+)
+
+type Key interface {
+	ToEtcdKey() string
+}
+
+type RosterKey struct {
+	types.NamespacedName
+}
+
+func (k RosterKey) ToEtcdKey() string {
+	return rosterKey + "/" + k.Namespace + "/" + k.Name
+}
+
+func (k RosterKey) String() string {
+	return k.ToEtcdKey()
+}
+
+type RosterEntryKey struct {
+	RosterKey
+	PlayerID uint
+}
+
+func (k RosterEntryKey) ToEtcdKey() string {
+	return rosterKey + "/" + k.Namespace + "/" + k.Name + "/" + strconv.Itoa(int(k.PlayerID))
+}
+
+func (k RosterEntryKey) String() string {
+	return k.ToEtcdKey()
+}
+
+const rosterKey = "/klyshko/roster"
+
+var etcdRosterKeyPattern = regexp.MustCompile("^" + rosterKey + "/(?P<namespace>(\\w|-)+)/(?P<jobName>(\\w|-)+)(?:/(?P<localPlayerID>\\d+))?$")
+
+func etcdKeyParts(s string) map[string]string {
```

Comment:
Because `etcdKeyParts` is just a utility method that splits the string into parts according to the pattern. Generating the concrete key instance is done in `ParseKey`.